modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx

```diff
@@ -0,0 +1,212 @@
+		});
+
+		fetch(addStyleBookEntryURL, {
+			body: objectToFormData(body),
+			method: 'POST',
+		})
+			.then((response) => response.json())
+			.then((responseContent) => {
+				if (responseContent.error) {
+					setLoading(false);
+					setErrorMessage(responseContent.error);
+				}
+				else if (responseContent.redirectURL) {
+					navigate(responseContent.redirectURL, {
+						beforeScreenFlip: closeModal,
+					});
+				}
+			})
+			.catch((response) =>
+				handleFormError(response)
+			);
+	};
+
+	const nameId = `${namespace}name`;
+	const themeIdId = `${namespace}tokenDefinition`;
+
+	return (
+		<>
+			<ClayModal.Header>
+				{Liferay.Language.get('add-style-book')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+
+				{errorMessage && (
+					<ClayAlert
+						displayType="danger"
+						onClose={() => {}}
+						title={Liferay.Language.get('error')}
+					>
+						{errorMessage}
+					</ClayAlert>
+				)}
+
+				<FieldBase
+					className="themeId"
+					helpMessage={Liferay.Language.get(
+						'the-style-book-will-be-created-based-on-the-selected-token-definition'
+					)}
+					id={themeIdId}
+					label={Liferay.Language.get('create-style-book-for')}
+				>
+					<Picker
+						active={active}
+						aria-labelledby="picker-label"
+						id={themeIdId}
+						onActiveChange={setActive}
+						onChange={(event: any) => {
+							const value = event.target.value;
+
+							setThemeId(frontendTokenDefinitionProviders[value]);
+						}}
+						onSelectionChange={setThemeId}
+						selectedKey={themeId.themeId}
+					>
+						{frontendTokenDefinitionProviders.map(
+							(frontendTokenDefinitionProvider) => (
+								<Option
+									key={frontendTokenDefinitionProvider.themeId}
+									textValue={frontendTokenDefinitionProvider.name}
+								>
+									{frontendTokenDefinitionProvider.name}
+								</Option>
+							)
+						)}
+					</Picker>
+				</FieldBase>
+
+				<ClayForm onSubmit={handleSubmit} ref={formRef}>
+					<div
+						className={`form-group ${
+							errorMessage ? 'has-error' : ''
+						}`}
+					>
+						<label
+							className="control-label"
+							htmlFor={`${namespace}${nameId}`}
+						>
+							{Liferay.Language.get('name')}
+
+							<span className="reference-mark">
+								<ClayIcon symbol="asterisk"/>
+							</span>
+						</label>
+
+						<ClayInput
+							id={nameId}
+							onChange={(event) => {
+								setName(event.target.value);
+
+								setErrorMessage(
+									event.target.value
+										? ''
+										: Liferay.Language.get(
+											'this-field-is-required'
+										)
+								);
+							}}
+							value={name}
+						/>
+					</div>
+				</ClayForm>
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton displayType="secondary" onClick={closeModal}>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+
+						<ClayButton
+							displayType="primary"
+							onClick={handleSubmit}
```

Comment:
instead of adding an onClick, we should add the [form attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#form) pointing to the id of the form. We should also add the attribute `type="submit"`.